src/platform/ESP32/ConfigurationManagerImpl.cpp

```diff
@@ -241,6 +241,28 @@ CHIP_ERROR ConfigurationManagerImpl::GetLocationCapability(uint8_t & location)
 #endif // CONFIG_ENABLE_ESP32_LOCATIONCAPABILITY
 }
 
+CHIP_ERROR ConfigurationManagerImpl::GetDeviceTypeId(uint32_t & deviceType)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DEVICE_TYPE
+    uint32_t value = 0;
+    CHIP_ERROR err = ReadConfigValue(ESP32Config::kConfigKey_PrimaryDeviceType, value);
+
+    if (err == CHIP_NO_ERROR)
+    {
+        deviceType = value;
+    }
+    else
+    {
+        deviceType = CHIP_DEVICE_CONFIG_DEVICE_TYPE;
+    }
+
+    return err;
+#else
+    deviceType = 0;
+    return CHIP_NO_ERROR;
```

Comment:
0 is not a valid value; why is it being returned with CHIP_NO_ERROR?

Should this just call the superclass to get whatever behavior that has?  Or should this whole declaration/implementation be conditioned on CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DEVICE_TYPE or something?

Reply:
Update